ml-agents/mlagents/trainers/components/reward_signals/gail/signal.py

```diff
@@ -42,6 +42,46 @@ def __init__(
             policy, 128, learning_rate, encoding_size, use_actions, use_vail
         )
         _, self.demonstration_buffer = demo_to_buffer(demo_path, policy.sequence_length)
+
```

Comment:
I think we need to do this for BCModule too, right?
https://github.com/Unity-Technologies/ml-agents/blob/a6500a665e7d6a1a8f7cfb21d5d7147d3f61ca71/ml-agents/mlagents/trainers/components/bc/module.py#L41

Maybe better to move it to a utility, or pass the expected `BehaviorSpec` to `demo_to_buffer()`?

Reply:
good point. I'll think on it.

Reply:
Ah, this ties back into the fact that the trainers still use BrainParameters to describe behaviors. As a stopgap could we just create a function that compares two BrainParameters and convert the demo's BehaviorSpec to a BrainParameters object using `behavior_spec_to_brain_parameters`?

Reply:
@ervteng thats basically what I did manually. I can move it into the BrainParameters class